src/main/java/io/github/thebusybiscuit/slimefun4/core/researching/Research.java

```diff
@@ -189,6 +195,43 @@ public Research addItems(ItemStack... items) {
         return items;
     }
 
+    /**
+     * Handle what to do when a {@link Player} clicks on an un-researched item in
+     * a {@link SlimefunGuideImplementation}.
+     *
+     * @author TheBusyBiscuit
+     * @author uiytt
+     *
+     * @param guide The {@link SlimefunGuideImplementation} used.
+     * @param player The {@link Player} who clicked on the item.
+     * @param profile The {@link PlayerProfile} of that {@link Player}.
+     * @param sfItem The {@link SlimefunItem} on which the {@link Player} clicked.
+     * @param category The {@link Category} where the {@link Player} was.
+     * @param page The page number of where the {@link Player} was in the {@link Category};
+     *
+     * @see ChestSlimefunGuide
+     * @see BookSlimefunGuide
+     */
+    @ParametersAreNonnullByDefault
+    public void guideClickInteraction(SlimefunGuideImplementation guide, Player player, PlayerProfile profile, SlimefunItem sfItem, Category category, int page) {
```

Comment:
This method name is absolutely horrible...
Please choose a better name, like `onPlayerResearch` or `unlockFromGuide`. But not this :NotLikeThis:

Reply:
I didn't know how to name it :sob: I asked on the discord several time but nobody answered :( 
Anyway, will change this